src/pk/asn1/der/utf8/der_encode_utf8_string.c

```diff
@@ -37,9 +37,12 @@ int der_encode_utf8_string(const wchar_t *in,  unsigned long inlen,
 
    /* get the size */
    for (x = len = 0; x < inlen; x++) {
-       if (in[x] < 0 || in[x] > 0x1FFFF) {
-          return CRYPT_INVALID_ARG;
-       }
+#if !defined(__WCHAR_MAX__) || __WCHAR_MAX__ > 0xFFFF
```

Comment:
as we have this ugly part of the code now twice and I don't see how to get rid of it, we should at least have it only once in the codebase.

how about adding `int der_utf8_valid_char(const wchar_t c)`?